src/commands/Birthday/set-birthday.ts

```diff
@@ -0,0 +1,95 @@
+import { getGuildMemberBirthDay, monthOfYearContainsDay, nextBirthday, TaskBirthDayData, yearIsLeap } from '#lib/birthday';
+import { Birthday } from '#lib/database/keys/settings/All';
+import { LanguageKeys } from '#lib/i18n/languageKeys';
+import { SkyraCommand } from '#lib/structures';
+import { GuildMessage } from '#lib/types';
+import { ApplyOptions } from '@sapphire/decorators';
+import { Args, CommandContext } from '@sapphire/framework';
+import { isNullish, Nullish } from '@sapphire/utilities';
+
+@ApplyOptions<SkyraCommand.Options>({
+	cooldown: 10,
+	description: LanguageKeys.Commands.Misc.SetBirthdayDescription,
+	extendedHelp: LanguageKeys.Commands.Misc.SetBirthdayExtended,
+	runIn: ['text']
+})
+export class UserCommand extends SkyraCommand {
+	public async run(message: GuildMessage, args: SkyraCommand.Args, context: CommandContext) {
+		const [birthdayRole, birthdayChannel, birthdayMessage] = await message.guild.readSettings([
+			Birthday.Role,
+			Birthday.Channel,
+			Birthday.Message
+		]);
+		if (this.isCorrectlyConfigured(birthdayRole, birthdayChannel, birthdayMessage))
+			return this.error(LanguageKeys.Commands.Misc.SetBirthdayNotConfigured, { prefix: context.commandPrefix });
+
+		const date = await this.handleArguments(args);
+
+		// delete any existing reminders
+		const currentTask = getGuildMemberBirthDay(message.guild.id, message.author.id);
+		if (currentTask) await this.context.client.schedules.remove(currentTask);
+
+		const birthday = nextBirthday(date.month, date.day);
+		await this.context.client.schedules.add('birthday', birthday, {
+			data: this.constructData(date, message)
+		});
+		return message.send(args.t(LanguageKeys.Commands.Misc.SetBirthdaySuccess, { nextBirthday: birthday.getTime() }));
+	}
+
+	private constructData(birthDate: DateWithOptionalYear, message: GuildMessage): TaskBirthDayData {
+		return {
+			guildID: message.guild.id,
+			userID: message.author.id,
+			year: birthDate.year,
+			month: birthDate.month,
+			day: birthDate.day
+		};
+	}
+
+	private async handleArguments(args: SkyraCommand.Args): Promise<DateWithOptionalYear> {
+		const result = await args.pickResult(UserCommand.dateWithOptionalYear);
+		if (result.success) return result.value;
+
+		const birthDate = await args.rest('date');
+
+		// The world's oldest human alive was born in January 1903:
+		if (birthDate.getTime() > Date.now() || birthDate.getFullYear() < 1903) this.error(LanguageKeys.Commands.Misc.SetBirthdayInvalidDate);
+
+		return { year: birthDate.getUTCFullYear(), month: birthDate.getUTCMonth() + 1, day: birthDate.getUTCDate() };
+	}
+
+	private isCorrectlyConfigured(birthdayRole: string | Nullish, birthdayChannel: string | Nullish, birthdayMessage: string | Nullish) {
+		// A birthday role, or a channel and message must be configured:
+		return !isNullish(birthdayRole) || (!isNullish(birthdayChannel) && !isNullish(birthdayMessage));
+	}
+
+	private static readonly currentYear = new Date().getUTCFullYear();
```

Comment:
This as a class field is.. not ok. What if someone runs the command on Dec 31st 2021 at 23:59 and then again at Jan 1st 00:00

Reply:
We reload the command. Either way, we have 13 years of margin before we have to accept newer years.